### PR TITLE
Sort the attributes for events

### PR DESCRIPTION
**Pull request by HackToday**
This is add support for #19559
We tried sort it in client side, and it sort follow go
sort : sorts a slice of strings in increasing order.

Signed-off-by: Kai Qiang Wu(Kennan) wkqwu@cn.ibm.com

### CODE REVIEW

**Comment by calavera** [comment]
Test is failing:

```
03:15:11 FAIL: docker_cli_events_test.go:142: DockerSuite.TestEventsContainerEventsAttrSort
03:15:11 
03:15:11 docker_cli_events_test.go:165:
03:15:11     c.Assert(matchedEvents, checker.Equals, 2)
03:15:11 ... obtained int = 4
03:15:11 ... expected int = 2
03:15:11 
```

Please, do NOT reorder the imports, use https://godoc.org/golang.org/x/tools/cmd/goimports to sort them automatically for you.

**Comment by HackToday** [comment]
hi @calavera  the import was used by 
`gofmat -s -w  <file>`

I did not manual change that

**Comment by HackToday** [comment]
Seems the server jenkins or event api have some issue, it could pass in my local container environment.

Strange ~

**Comment by HackToday** [comment]
hi, @thaJeztah  @calavera  do you know what's `janky` about ?

I found same unit test, can pass in experimental
03:19:03 PASS: docker_cli_events_test.go:142: DockerSuite.TestEventsContainerEventsAttrSort	0.553s

But failed for `janky`

**Comment by calavera** [comment]
> the import was used by 
> gofmat -s -w <file>

no idea why it's changing the order, but we're not going to merge that change :wink: 

The test is getting more events in the janky suite, that's why it's failing.

File: integration-cli/docker_cli_events_test.go [hunk 1]
```diff
@@ -139,6 +139,32 @@ func (s *DockerSuite) TestEventsContainerEvents(c *check.C) {
...
+	out, _ := dockerCmd(c, "events", fmt.Sprintf("--since=%d", since), fmt.Sprintf("--until=%d", daemonTime(c).Unix()))
+	events := strings.Split(out, "\n")
+	events = events[:len(events)-1]
+
+	nEvents := len(events)
+	c.Assert(nEvents, checker.GreaterOrEqualThan, 5) //Missing expected event
+	matchedEvents := 0
+	for _, event := range events {
+		matches := parseEventText(event)
+		if matches["eventType"] == "container" && matches["action"] == "create" {
```
Comment (calavera) [inline integration-cli/docker_cli_events_test.go]: check the id of the container here to make sure we get the events for the container you just created.

Comment (HackToday) [inline integration-cli/docker_cli_events_test.go]: Sure @calavera  I also noticed docker code now `import case` not follow alphabetic  order,

I could not know what's go rule for docker ? 

Perhaps that way to confirm is I just insert a `import case` and then check docker jenkins failed. And get that what's the right change

**Comment by HackToday** [comment]
Not know why jenkins hang there

**Comment by cpuguy83** [comment]
@HackToday Probably because of yesterday's GH outage...

**Comment by HackToday** [comment]
Hi @thaJeztah  how can I kick a run again ? It seems the jenkins tests can not run(it has been waited for one day)

**Comment by thaJeztah** [comment]
I'll kick it; Janky was having issues due to the GitHub outage

**Comment by HackToday** [comment]
It looks like remote system is not stable

```
05:25:35  ---> Running in ebeaead39ce1
05:25:35 + mktemp -d
05:25:35 + export GOPATH=/tmp/tmp.84bumRWIaw
05:25:35 + git clone https://github.com/docker/distribution.git /tmp/tmp.84bumRWIaw/src/github.com/docker/distribution
05:25:35 Cloning into '/tmp/tmp.84bumRWIaw/src/github.com/docker/distribution'...
05:25:51 error: RPC failed; result=52, HTTP code = 0
05:25:51 fatal: The remote end hung up unexpectedly
05:26:08 Removing intermediate container ebeaead39ce1
05:26:08 The command '/bin/sh -c set -x 	&& export GOPATH="$(mktemp -d)" 	&& git clone https://github.com/docker/distribution.git "$GOPATH/src/github.com/docker/distribution" 	&& (cd 

```

File: api/client/events.go [hunk 2]
```diff
@@ -99,7 +100,14 @@ func printOutput(event eventtypes.Message, output io.Writer) {
 
 	if len(event.Actor.Attributes) > 0 {
 		var attrs []string
-		for k, v := range event.Actor.Attributes {
+		var keys []string
+		var v string
```
Comment (calavera) [inline api/client/events.go]: this variable doesn't need to be declared here, use `v := event.Actor.Attributes` in line 110.

**Comment by HackToday** [comment]
Still the build is not stable, checked the above output

```

13:16:31 Build step 'Execute shell' marked build as failure
13:16:31 Performing Post build task...
13:16:31 Could not match :Build timed out  : False
13:16:31 Logical operation result is FALSE
13:16:31 Skipping script  : cd c:\scripts
13:16:31 sh timeout-post-build.sh
```

File: integration-cli/docker_cli_events_test.go [hunk 3]
```diff
@@ -140,6 +140,35 @@ func (s *DockerSuite) TestEventsContainerEvents(c *check.C) {
 	c.Assert(containerEvents[4], checker.Equals, "destroy", check.Commentf(out))
 }
 
+func (s *DockerSuite) TestEventsContainerEventsAttrSort(c *check.C) {
+	testRequires(c, DaemonIsLinux)
```
Comment (jhowardmsft) [inline integration-cli/docker_cli_events_test.go]: Shouldn't be a need for this testRequires following merge of https://github.com/docker/docker/pull/19839

**Comment by HackToday** [comment]
@jhowardmsft  still windows jenkins failed for not related with my change.

Could any docker maintainers to kick the test again ?   And BTW How can others(for example me)  kick it ?

**Comment by thaJeztah** [comment]
@HackToday restarted the build. Restarting the builds is currently only possible for maintainers that are voiced in the #docker-maintainers IRC channel

**Comment by HackToday** [comment]
ping @calavera @vieux   PTAL. Thanks

**Comment by calavera** [comment]
LGTM

**Comment by vieux** [comment]
LGTM

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**